src/scripts/main.js

```diff
@@ -1 +1,53 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+const firstPromise = new Promise((resolve, reject) => {
+  body.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+  setTimeout(() => reject(new Error(`First promise was rejected`)), 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  body.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  body.addEventListener('contextmenu', (ev) => {
+    ev.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = Promise.all([
+  new Promise(resolve => {
+    body.addEventListener('click', () => {
+      resolve('Third promise was resolved');
+    });
+  }),
+
+  new Promise(resolve => {
+    body.addEventListener('contextmenu', (ev) => {
+      ev.preventDefault();
+      resolve();
+    });
+  }),
+]);
+
+firstPromise
+  .then(message => getMessage('success', message))
+  .catch(error => getMessage('warning', error));
+
+secondPromise
+  .then(message => getMessage('success', message));
+
+thirdPromise.then(message => getMessage('success', message));
+
+function getMessage(classOfMessage, message) {
```

Comment:
In most cases we can easily avoid using redundant `of` to make our code cleaner
```suggestion
function getMessage(messageClass, message) {
```